File: src/main/java/com/skocur/imagecipher/plugin/FileLister.java
package com.skocur.imagecipher.plugin;

import java.io.File;
import java.net.URI;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class FileLister {

  @NotNull
  public static List<URI> listUris(File[] jars) {
    return Arrays.stream(jars)
        .map(File::toURI)
        .collect(Collectors.toList());
  }

  @Nullable
  public static File[] listJars(String pluginsPath) {
    File dir = new File(pluginsPath);
    return dir.listFiles((directory, name) -> name.endsWith(".jar"));
  }
}

File: app/src/main/kotlin/com/imagecipher/app/plugin/FileLister.kt
package com.imagecipher.app.plugin

import java.io.File
import java.net.URI

object FileLister {

    fun listUris(jars: Array<File>): List<URI> {
        return jars.map { it.toURI() }
    }

    fun listJars(pluginsPath: String): Array<File>? {
        val dir = File(pluginsPath)
        return dir.listFiles { _, name -> name.endsWith(".jar") }
    }
}
